sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -1866,7 +1901,16 @@ private Mono<ResourceResponse<Document>> createDocumentInternal(String collectio
                 options, disableAutomaticIdGeneration, OperationType.Create);
 
             Mono<RxDocumentServiceResponse> responseObservable =
-                requestObs.flatMap(request -> create(request, requestRetryPolicy, getOperationContextAndListenerTuple(options)));
+                requestObs.flatMap(request -> {
+                    CosmosEndToEndOperationLatencyPolicyConfig endToEndPolicyConfig = getEndToEndOperationLatencyPolicyConfig(options);
```

Comment:
Does the timeout policy will be only applied on document related requests? or it will expand into other operation type? (like read collection, read database etc)

Reply:
Had the same question/thought - but IMO we should keep this to data plane operations - point operations, queries, possibly bulk and batch (including all helper requests like for query plan) - but this needs to be called out much more clearly in the PR description.
Everything else should never happen (read database, collection etc.) should never happen on the hot path anyway in a customer's application.

Reply:
clarify a little bit: so for example read item operation, internally we might need to read collection, populate cache etc, so these internal flows will still be capped under the e2e timeout right?

Reply:
no, they will not be, because metadata requests will happen in background now, no?

Reply:
It will only be enabled for Document operations. This call here is only called by createDocument